scripts/prohibit-direct-merges-to-master

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+self_dir=$(cd "$(dirname "$0")" &>/dev/null; pwd -P)
+source "$self_dir"/utils.sh
+
+current_branch=$(git branch --show-current)
+num_of_prs_from_current_to_master=$(gh pr list --state open --base master --head "$current_branch" | wc -l) # one line per one PR
+
```

Comment:
Also, note that if there's no PR, then `gh pr list --state open --base master --head ...` still prints out the output... albeit to stderr, not stdout. Still, to make sure we don't depend on the format of the output (as `gh pr list` is rather a **porcelain** command), let's use sth like:

```shell
gh pr list --state open --base master --head "$current_branch" --json number --jq length
```

Reply:
I did a little test on that locally. Looks like current logic is OK. If specified pull request really exists - the result of the command is `1`. If it does not exist - the result is `0`. Output from command line:
```
lkontowski@lkontowski-MacBook-Pro /Users/lkontowski/code/git-machete-intellij-plugin [develop]
% gh pr list --state open --base master --head prevent_merging_branches_to_master_accidentally | wc -l
       0
lkontowski@lkontowski-MacBook-Pro /Users/lkontowski/code/git-machete-intellij-plugin [develop]
% gh pr list --state open --base develop --head prevent_merging_branches_to_master_accidentally | wc -l
       1
lkontowski@lkontowski-MacBook-Pro /Users/lkontowski/code/git-machete-intellij-plugin [develop]
% gh pr list --state open --base INVALID_BRANCH_NAME --head prevent_merging_branches_to_master_accidentally | wc -l
       0
lkontowski@lkontowski-MacBook-Pro /Users/lkontowski/code/git-machete-intellij-plugin [develop]
% gh pr list --state open --base develop --head INVALID_BRANCH_NAME | wc -l                                        
       0
```

Do you think that using `json` format for output is neede here?

Reply:
Okay, AFAICS the output when piped to the process is different and less porcelain then when sent straight to the terminal:

```
$ gh pr list --state open --base develop --head prevent_merging_branches_to_master_accidentally | cat
1169	Prevent merging branches to master accidentally	prevent_merging_branches_to_master_accidentally	OPEN	2022-09-26 13:26:48 +0000 UTC

$ gh pr list --state open --base develop --head prevent_merging_branches_to_master_accidentally 

Showing 1 of 1 pull request in VirtusLab/git-machete-intellij-plugin that matches your search

#1169  Prevent merging branches to master accidentally  prevent_merging_branches_to_master_accidentally  about 20 hours ago
```

So maybe indeed, in the non-terminal version, it's safe to use as is

Reply:
The output sent to terminal looked too porcelain/brittle to me

Reply:
Ok... Now I am a little confused. 
> So maybe indeed, in the non-terminal version, it's safe to use as is

Do I get it right that I can leave it as it is?
`num_of_prs_from_current_to_master=$(gh pr list --state open --base master --head "$current_branch" | wc -l)`

Or should I change it so that it uses `json` format as "The output sent to terminal looked too porcelain/brittle to me"?

Reply:
Yes, pls leave as is